plugins/modules/clustering/pacemaker/pacemaker_cluster.py

```diff
@@ -0,0 +1,369 @@
+
+        match_object = re.search(r'Corosync Nodes:\n((?:\s\S+)*?)\n', out)
+        corosync_nodes = match_object.group(1).strip().split(' ')
+        match_object = re.search(r'Pacemaker Nodes:\n((?:\s\S+)*?)\n', out)
+        pacemaker_nodes = match_object.group(1).strip().split(' ')
+        # only include nodes in both corosync_nodes and pacemaker_nodes
+        existing_nodes = [node for node in corosync_nodes if node in pacemaker_nodes]
+        for existing_node in existing_nodes:
+            if existing_node in nodes:
+                nodes.remove(existing_node)
+        # if there are still nodes unaccounted for, add them to the cluster
+        if len(nodes) > 0:
+            changed = True
+            authenticate_nodes(module, nodes, pcs_user, pcs_password)
+            for node in nodes:
+                cmd = "pcs cluster node add --start %s" % node
+                rc, out_node, err = module.run_command(cmd)
+                if rc == 1:
+                    module.fail_json(msg="Failed to add node to cluster.\nCommand: `%s`\nError: %s" % (cmd, err))
+
+        # now that all nodes should be present, check their status
+        online_nodes, offline_nodes = get_nodes_status(module)
+
+        # check if properties already exist
+        match_object = re.search(r'Cluster Properties:\n (.*?:.*?\n)*', out)
+        existing_properties = re.findall(r'(\S*?): (\S*?)\n', match_object.group(0))
+        for existing_property in existing_properties:
+            if existing_property in properties:
+                properties.remove(existing_property)
+
+        if len(properties) > 0:
+            changed = True
+            cmd = "pcs property set %s" % " ".join(["%s=%s" % props for props in properties.items()])
+            rc, out, err = module.run_command(cmd)
+            if rc == 1:
+                # if the cluster is up but still failed to get cluster config => error
+                module.fail_json(msg="Failed to set cluster properties.\nCommand: `%s`\nError: %s" % (cmd, err))
+
+        return changed
+
+
+def clean_cluster(module, timeout):
+    cmd = "pcs resource cleanup"
+    rc, out, err = module.run_command(cmd)
+    if rc == 1:
+        module.fail_json(msg="Command execution failed.\nCommand: `%s`\nError: %s" % (cmd, err))
+
+
+def main():
+    argument_spec = dict(
+        state=dict(type='str', choices=['online', 'offline', 'restart', 'cleanup', 'present', 'absent']),
+        name=dict(type='str', required=True),
+        pcs_user=dict(type='str', required=True),
+        pcs_password=dict(type='str', required=True),
+        timeout=dict(type='int', default=300),
+        force=dict(type='bool', default=True),
+        nodes=dict(type='list', elements='str'),
+        properties=dict(type='dict', default={}),
+    )
+
+    module = AnsibleModule(
+        argument_spec,
+        supports_check_mode=True,
```

Comment:
Where in the code you are using `module.check_mode` to make sure that changes are only applied when not in check mode?